src/screens/profile_details/hooks.ts

```diff
@@ -45,12 +45,20 @@ export const useProfileDetails = () => {
     if (!regexCheck || !configProfile) {
       router.replace('/');
     }
-
     if (configProfile) {
       fetchDesmosProfile(R.pathOr('', ['query', 'dtag'], router));
     }
   }, [R.pathOr('', ['query', 'dtag'], router)]);
 
+  useEffect(() => {
+    if (state.desmosProfile) {
+      const dtagInput = router.query.dtag as string;
+      if (state.desmosProfile.dtag.toUpperCase() !== dtagInput.toUpperCase()) {
+        router.push({ pathname: `/@${state.desmosProfile.dtag}` }, `/@${state.desmosProfile.dtag}`, { shallow: true });
+      }
+    }
+  }, [state.desmosProfile]);
+
```

Comment:
1) dtagInput will never == state.desmosProfile.dtag because state.desmosProfile.dtag will be `Yaya` but dtagInput will be `@Yaya`

2) we only want to replace **if** it is a casing mismatch. We won't want to replace if one is `@Dogs` and the one is `Cats`